airbyte-commons-temporal/src/main/java/io/airbyte/commons/temporal/scheduling/RouterService.java

```diff
@@ -37,9 +45,20 @@ public String getTaskQueue(final UUID connectionId, final TemporalJobType jobTyp
     return taskQueueMapper.getTaskQueue(geography, jobType);
   }
 
+  // This function is only getting called for discover/check functions. Today (02.07) they are behind
+  // feature flag
+  // so even the geography might be in EU they will still be directed to US.
   public String getTaskQueueForWorkspace(final UUID workspaceId, final TemporalJobType jobType) throws IOException {
-    final Geography geography = configRepository.getGeographyForWorkspace(workspaceId);
-    return taskQueueMapper.getTaskQueue(geography, jobType);
+    if (!WORKSPACE_ROUTING_JOB_TYPE_SET.contains(jobType)) {
+      throw new RuntimeException("Jobtype not expected to call - getTaskQueueForWorkspace - " + jobType);
+    }
+
+    if (featureFlags.routeTaskQueueForWorkspaceEnabled() || featureFlags.routeTaskQueueForWorkspaceAllowList().contains(workspaceId.toString())) {
```

Comment:
Should this logic be an `&&` instead of a `||`? Or is the first feature flag meant to enable the behavior for everyone, and the second feature flag is meant for enabling it just for specific workspaces?

Reply:
I think I answered my own question, they're two separate feature flags, and the first one will default to false until we're ready to enable the behavior for everyone

Reply:
Yes! The experiment phase will mark the first flag to false but to include a few workspaces of our own.